lib/pure/memfiles.nim

```diff
@@ -245,6 +249,30 @@ proc open*(filename: string, mode: FileMode = fmRead,
       if close(result.handle) == 0:
         result.handle = -1
 
+proc flush*(f: var MemFile; attempts: int = 3) =
+  ## Flushes `f`'s buffer for the number of attempts equal to `attempts`.
+  ## If were errors an exception `OSError` will be raised.
+  when defined(windows):
+    var retry = 0
+    var res = false
+    while retry <= attempts:
+      res = flushViewOfFile(f.mem, 0) != 0
+      if res:
```

Comment:
Also, you're retrying no matter what the error is, that's bad. Only retry on the ``ERROR_LOCK_VIOLATION`` error!

Reply:
I don't see why replicating this to POSIX makes sense. In fact, I'm skeptical about whether this should be configurable via a param at all.